forge/csrc/passes/commute_utils.cpp

```diff
@@ -1282,15 +1295,12 @@ bool try_commute_bcast_through_clone(graphlib::Graph *graph, graphlib::OpNode *n
 
                 int operand_dim_size = operand->shape()[dim];
                 // Hoist the transpose before the bcasts
-                if (dim == -1 or dim == -2)
-                {
-                    erase_tms.push_back(i);
-                    dim = (dim == -1) ? -2 : -1;  // flip the dim
-                    updated_shape[dim] = operand_dim_size;
-                    std::get<int>(op_type.attr[0]) = dim;
-                    for (auto user_edge : graph->user_data_edges(op))
-                        graph->get_edge_attributes(user_edge)->prepend_tm(op_type);
-                }
+                erase_tms.push_back(i);
```

Comment:
Let's add comment (s) explaining following logic

Reply:
Since all logic is in `update_bcast_dim_commuted_through_transpose` added coments to explain how we get new broadcast dim...